### ISSUE TITLE
A config option to ignore rules for specific lines by pattern matching or something similar

### ISSUE THREAD

**Issue by Lassii-** [astral-sh/ruff#8687]
Hello,

I'm working in a project where we're currently introducing Ruff as the solution for linting and it's overall working great but we have one situation where I haven't been able to find a good solution. We use Python and Robot Framework and we define the RF test keywords to be used in RF in Python, for example `@keyword("User presses the info button"` and some of these lines are longer than 120 characters which is our line length limit set in Ruff. 

However for these lines we would like to ignore the rule because the readability gets worse if they're split. I'm aware that you can add `# noqa: E501` to a line to ignore the rule but this is quite cumbersome and needs manual maintenance plus these files also have other lines that we want the rule to be enabled for so full file E501 ignore is not good. 

There doesn't currently seem to be a way to match lines inside files by a pattern to ignore specific rules, for example lines that start with the already mentioned `@keyword` so could something like this be implemented?

Thank you,
Lassi

**Comment by zanieb** [comment]
Interesting. I could see a regex -> rule codes mapping to ignore violations where the diagnostic range matches that in the regex. In practice though, I think this would be hard to implement in a performant way that matches user expectations.

**Comment by MusicalNinjaDad** [comment]
+1 for this idea with a different use case:

I find I start all my tests with:
```py
import pytest  # noqa: F401, RUF100
```

so that I can temporarily add and remove various marks etc. (e.g. `pytest.mark.xfail` or `pytest.mark.skip`) as tests and code changes, without constantly reimporting pytest and removing the import if I remove the mark.

I'd rather not add `F401` to my
```toml
[tool.ruff.lint.per-file-ignores]
    # Additional ignores for tests 
    "**/test_*.py" = [
     ...
    ]
```
as I want other imports to be checked.

**Comment by coredumperror** [comment]
I'd also like a feature like this, to make Ruff ignore my `nosemgrep` comments.

For instance, it'll complain about the following as "commented out code" (rule ERA001):

     # nosemgrep:direct-use-of-jinja2

That's a false positive in this particular case, though I would rather not ignore the rule globally. But it's also rather cumbersome to add `# noqa: ERA001` to every single one of my nosemgrep comments.

Some way to say "Ignore ERA001 on all lines that begin with `# nosemgrep`" would be great.

**Comment by MichaReiser** [comment]
@coredumperror You can add `nosemgrep` to the [`task-tags`](https://docs.astral.sh/ruff/settings/#lint_task-tags) setting. While not intended for this use case, it should work.

**Comment by KonWys01** [comment]
I would also love to get this feature. In my case im trying to filter out imports that start with "from project.application..." I want this rule to be filtering these imports inside /project/application/* as I prefere locale imports when im in the app (instead of absolute imports).

I know i can use ruff.toml inside /project/application/ to apply rules to only files in this directory, but without regex functionality its not possible

**Comment by MichaReiser** [comment]
@KonWys01 I think a better solution to your problem is if we extend `per-file-ignores` to allow overriding arbitrary settings AND add glob support to `banned-from`

**Comment by KonWys01** [comment]
@MichaReiser Were you already planning on implementing such feature or its just an idea for now?

**Comment by MichaReiser** [comment]
Allowing more flexible `per-file-ignores` is something I think we should be doing but isn't on our immediate roadmap. I believe there's an existing issue for allowing globs/regex in more places related to imports but this is also something that isn't on our immediate roadmap. 

I'm more outlining that I don't think your use case requires the feature request by the issue author.
